sormas-app/app/src/main/java/de/symeda/sormas/app/pathogentest/list/PathogenTestListViewModel.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * SORMAS® - Surveillance Outbreak Response Management & Analysis System
+ * Copyright © 2016-2018 Helmholtz-Zentrum für Infektionsforschung GmbH (HZI)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package de.symeda.sormas.app.pathogentest.list;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+import androidx.paging.DataSource;
+import androidx.paging.LivePagedListBuilder;
+import androidx.paging.PagedList;
+import androidx.paging.PositionalDataSource;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import de.symeda.sormas.app.backend.common.DatabaseHelper;
+import de.symeda.sormas.app.backend.sample.PathogenTest;
+import de.symeda.sormas.app.backend.sample.Sample;
+
+public class PathogenTestListViewModel extends ViewModel {
```

Comment:
Please either use the paged list model properly (i.e. implement the loadInitial and loadRange methods like in the other view models to make the list lazy loading) OR make the list eager in the first place, i.e. don't use a paged list. Currently, this is a lazy list that simulates eager loading which is not a good concept.

Reply:
I need clarification

1. "use the paged list model properly" 
Need to add DatabaseHelper.getPathogenTestDao()
Need to implement queryByCriteria() method in PathogenTestDao

2. "make the list eager in the first place"
PathogenTestListAdapter extends BindingPagedListAdapter which is a paged list adapter. So what is the eager loading list adapter class to extend

Reply:
I don't think we use any eager lists anymore in the app, so the first suggestion is probably the easier solution. Please have a look especially at the loadInitial and loadRange methods of other view models. And yes, you're right, you'll need to add queryByCriteria and countByCriteria methods.